### PR TITLE
Add 'IsVisible` query class for UI Tester

### LINKED ISSUE #1291: Add 'IsVisible` query class for UI Tester
In some cases, an application/ui may have certain parts of the UI change between visible and not visible depending on the state of the UI.  It is likely that this behavior is something that one would want to verify with tests.

Note that all traitsui editors have a `visible` trait, but also toolkit specific elements (e.g. QtGui.Qwidgets) have `isVisible` methods.

### PR DESCRIPTION
**Pull request by aaronayres35**
closes #1291 

Now that #1490 has been merged this process was quite simple :)

This PR makes it so that now any UIWrapper object wrapping a target with a control that is either a QtWidget or wx.Window can now support an `IsVisible` query. A comparable thing was done for `IsEnabled` in 1490.

This PR also updates the relevant tests for `_control_widget_registry` and adds a test for the `visible_when.py` demo.

### CODE REVIEW

**Comment by aaronayres35** [comment]
#1551 has returned! this time with full traceback

<details>

```
test_base_url_changed (traitsui.tests.editors.test_html_editor.TestHTMLEditor) ... Fatal Python error: Illegal instruction

Current thread 0x00007feb44de9740 (most recent call first):
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/traitsui/testing/_gui.py", line 36 in process_cascade_events
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/traitsui/testing/tester/ui_tester.py", line 116 in create_ui
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/contextlib.py", line 88 in __exit__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/traitsui/tests/editors/test_html_editor.py", line 253 in test_base_url_changed
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/case.py", line 605 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/case.py", line 653 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/runner.py", line 176 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/main.py", line 256 in runTests
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/main.py", line 95 in __init__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/__main__.py", line 18 in <module>
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/execfile.py", line 183 in run_python_file
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/execfile.py", line 113 in run_python_module
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/cmdline.py", line 627 in do_run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/cmdline.py", line 483 in command_line
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/cmdline.py", line 756 in main
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/__main__.py", line 8 in <module>
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/runpy.py", line 85 in _run_code
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/runpy.py", line 193 in _run_module_as_main
Command '['edm', 'run', '-e', 'traitsui-test-3.6-pyside2', '--', 'python', '-X', 'faulthandler', '-W', 'default', '-m', 'coverage', 'run', '-p', '-m', 'unittest', 'discover', '-v', 'traitsui']' returned non-zero exit status 252.
/usr/bin/bash /home/runner/work/_actions/GabrielBB/xvfb-action/v1/cleanup.sh
Killing the following xvfb processes: 4894
Error: The process '/usr/bin/xvfb-run' failed with exit code 1
```

</details>

**Comment by aaronayres35** [comment]
It looks like #1553 may have been successful, no longer seeing 1551 failure

File: traitsui/testing/tester/_ui_tester_registry/qt4/_control_widget_registry.py [hunk 1]
```diff
@@ -32,6 +32,19 @@ def _handle_is_enabled(wrapper, interaction):
     return wrapper._target.control.isEnabled()
 
 
+def _handler_is_visible(wrapper, interaction):
```
Comment (rahulporuri) [inline traitsui/testing/tester/_ui_tester_registry/qt4/_control_widget_registry.py:35]: nit - conform to the existing naming convention
```suggestion
def _handle_is_visible(wrapper, interaction):
```

File: traitsui/testing/tester/_ui_tester_registry/qt4/_control_widget_registry.py [hunk 2]
```diff
@@ -63,5 +76,6 @@ def get_widget_registry():
         can_support=_is_target_control_a_qt_widget,
         interaction_to_handler={
             IsEnabled: _handle_is_enabled,
+            IsVisible: _handler_is_visible,
```
Comment (rahulporuri) [inline traitsui/testing/tester/_ui_tester_registry/qt4/_control_widget_registry.py:79]: if the above suggestion is accepted

```suggestion
            IsVisible: _handle_is_visible,
```

File: traitsui/testing/tester/_ui_tester_registry/wx/tests/test_control_widget_registry.py [hunk 3]
```diff
@@ -58,10 +58,14 @@ def test_is_enabled(self):
         self.widget.Enable(True)
         self.assertTrue(self.good_wrapper.inspect(IsEnabled()))
 
+    def test_is_visible(self):
+        self.widget.Show(True)
+        self.assertTrue(self.good_wrapper.inspect(IsVisible()))
+
```
Comment (rahulporuri) [inline traitsui/testing/tester/_ui_tester_registry/wx/tests/test_control_widget_registry.py:64]: Let's add the test invisible here - like what we have in qt - unless there is a reason we can't.

**Review by rahulporuri — APPROVED** [review]
LGTM

**Review by jwiggins — APPROVED** [review]
LGTM

**Comment by aaronayres35** [comment]
Looks like another occurrence of #1551 
```
test_base_url_changed (traitsui.tests.editors.test_html_editor.TestHTMLEditor) ... Fatal Python error: Illegal instruction

Current thread 0x00007f0f6474c740 (most recent call first):
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/traitsui/testing/_gui.py", line 36 in process_cascade_events
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/traitsui/testing/tester/ui_tester.py", line 116 in create_ui
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/contextlib.py", line 88 in __exit__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/traitsui/tests/editors/test_html_editor.py", line 253 in test_base_url_changed
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/case.py", line 605 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/case.py", line 653 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 122 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/suite.py", line 84 in __call__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/runner.py", line 176 in run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/main.py", line 256 in runTests
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/main.py", line 95 in __init__
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/unittest/__main__.py", line 18 in <module>
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/execfile.py", line 183 in run_python_file
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/execfile.py", line 113 in run_python_module
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/cmdline.py", line 627 in do_run
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/cmdline.py", line 483 in command_line
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/cmdline.py", line 756 in main
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/site-packages/coverage/__main__.py", line 8 in <module>
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/runpy.py", line 85 in _run_code
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside2/lib/python3.6/runpy.py", line 193 in _run_module_as_main
Command '['edm', 'run', '-e', 'traitsui-test-3.6-pyside2', '--', 'python', '-X', 'faulthandler', '-W', 'default', '-m', 'coverage', 'run', '-p', '-m', 'unittest', 'discover', '-v', 'traitsui']' returned non-zero exit status 252.
/usr/bin/bash /home/runner/work/_actions/GabrielBB/xvfb-action/v1/cleanup.sh
No xvfb processes to kill
Error: The process '/usr/bin/xvfb-run' failed with exit code 1
```

**Pull request merged**